File: Numbers/binary_to_decimal/binary_to_decimal.java
import java.util.Scanner;

class binary_to_decimal {
    public static void main(String[] args) {
        System.out.print("Enter a binary number: "); /* This is such a lazy solution. */
        System.out.println(Integer.parseInt(new Scanner(System.in).nextLine(), 2));
    }
}

File: Numbers/binary_to_decimal/binary_to_decimal.kt
import java.util.*

fun main(){
    val input = Scanner(System.`in`)

    //Binary to Decimal
    //Using input
    println("Enter Binary to Convert to Decimal")
    println(input.next().asDecimal())

    //Direct usage
    println("1011".asDecimal())

    //Decimal to Binary
    //Using input
    println("Enter Integer to Convert to Binary")
    println(input.nextInt().asBinary())

    //Direct usage
    println(1024.asBinary())
}

fun String.asDecimal() = try{ "${this.toInt(2)}" } catch (e:NumberFormatException){ "Not a valid Input" }
fun Int.asBinary():String = Integer.toBinaryString(this)
